elastic/kibana
# [Logs UI] Set Logs stream time ruler according to set timezone

## 📓 Summary

Closes #94267 

This change fixes the current mismatch between the TimeRuler bar on the Log Stream feature and the set timezone in the advanced option.

I didn't find a more elegant way to implement it (dates and time zones are always tricky), so any better approach is welcome! 

The current solution consists of:
In order to get the correct timestamp for a custom timezone
1. Start from the current start/end timestamp
2. Get the current timezone offset in milliseconds (positive or negative)
3. Get the UTC offset in milliseconds (positive or negative)
4. Sum the previously retrieved elements all together.

This guarantees the timestamp we return is related to the timezone set globally in moment-js.

### Example
**Timezone:** US/Eastern ==> -300mins UTC offset
**Local position:** GMT+1(Madrid) -60mins timezone offset

Current timestamp + (-300 * 60000) + (-60 * 60000) = timestamp for US/Eastern

## 🧪 Testing

- Navigate to `Stack Management` -> `Advanced Settings` -> Search `time zone`
- Set the timezone to any preferred location
- Navigate to `Observability` -> `Logs` -> `Stream`
- Verify the logs entries timestamp and the TimeRuler range are in sync
